adr/0026-schema.md

```diff
@@ -0,0 +1,614 @@
+        path: replicaCount
+    # Everything below this line is changing https://github.com/defenseunicorns/zarf/issues/2245  
+    helm: # Only one of helm, git, oci, url, or local is allowed
+      url: https://stefanprodan.github.io/podinfo
+      name: podinfo # replaces repoName since it's only applicable in this situation
+    git:
+      url: https://stefanprodan.github.io/podinfo
+      path: charts/podinfo
+    oci:
+      url: oci://ghcr.io/stefanprodan/charts/podinfo
+    local:
+      path: chart
+  dataInjections:
+  - source: zim-data
+    target:
+      namespace: my-namespace
+      selector: app=my-app
+      container: data-loader
+      path: /data
+    compress: true
+  files:
+  - source: source-file.txt
+    target: target-file.txt
+    shasum: shasum
+    executable: false
+    symlinks:
+    - /path/to/symlink
+    extractPath: /path/to/extract
+  images:
+  - podinfo@v1
+  repos:
+  - https://github.com/defenseunicorns/zarf
+  extensions:
+    bigbang:
+      version: bbVersion
+      repo: https://repo1.com/mybbrepo
+      valuesFiles:
+      - values.yaml
+      skipFlux: false
+      fluxPatchFiles:
+      - flux-patch.yaml
+  actions:
+    onCreate:
+      defaults:
+        mute: true
+        maxTotalSeconds: 0
+        maxRetries: 0
+        dir: dir
+        env:
+        - ENV_VAR=FOO
+        shell:
+          darwin: sh
+          linux: sh
+          windows: powershell
+      before:
+      - mute: false
+        maxTotalSeconds: 0
+        maxRetries: 0
+        dir: dir
+        env:
+        - ENV_VAR=FOO
+        cmd: echo hello
+        shell:
+          darwin: sh
+          linux: sh
+          windows: powershell
+        setVariables:
+        - name: VAR
+          sensitive: false
+          autoIndent: true
+          pattern: ".+"
+          type: raw
+        description: action-description
+        wait:
+          cluster: # Only one of cluster / network can be used
+            kind: pod
+            name: my-pod
+            namespace: pod-ns
+            condition: ready
+          network:
+            protocol: http
+            address: github.com
+            code: 200
+      after:
+      - mute: false
+        maxTotalSeconds: 0
+        maxRetries: 0
+        dir: dir
+        env:
+        - ENV_VAR=FOO
+        cmd: echo hello
+        shell:
+          darwin: sh
+          linux: sh
+          windows: powershell
+        setVariables:
+        - name: VAR
+          sensitive: false
+          autoIndent: true
+          pattern: ".+"
+          type: raw
+        description: action-description
+        wait:
+          cluster: # Only one of cluster / network can be used
+            kind: pod
+            name: my-pod
+            namespace: pod-ns
+            condition: ready
+          network:
+            protocol: http
+            address: github.com
+            code: 200
+      onSuccess:
+      - mute: false
+        maxTotalSeconds: 0
+        maxRetries: 0
```

Comment:
I think changing to `retries` would be better than `maxRetries`

Reply:
Agree